src/openforms/formio/api/views.py

```diff
@@ -59,13 +63,19 @@ def post(self, request, *args, **kwargs):
                 content_type="text/plain",
             )
 
+        submission = serializer.validated_data["submission"]
+
+        if not owns_submission(request, submission.uuid):
+            raise PermissionDenied()
```

Comment:
additionally, this doesn't result in an response with `text/plain` content type, which is required for Formio upload fields.